### ISSUE TITLE
west: spdx: Blank FileChecksum field for missing build file

### ISSUE THREAD

**Issue by swinslow** [zephyrproject-rtos/zephyr#42072]
**Describe the bug**

The `west spdx` functionality uses the output from CMake's file-based API to determine which build files are created from the various build targets. It then checks each built file to get its checksum and other details for the `build.spdx` SPDX Document.

Using the current main branch to build some basic samples (e.g. samples/hello_world, samples/basic/blinky), there is now a file which is being reported by the CMake metadata as being built: `build/zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a`.

However, when `west spdx` tries to gather this file's checksum after the build is complete, this file is not present. As a result, `west spdx` includes the following line:
```
FileChecksum: SHA1: 
```

which causes the SPDX Document to be invalid due to the missing checksum after `SHA`:`.

**To Reproduce**

1. run `west build` followed by `west spdx` as described in the [Zephyr documentation](https://docs.zephyrproject.org/latest/guides/west/zephyr-cmds.html#software-bill-of-materials-west-spdx)
2. try to [validate](https://tools.spdx.org/app/validate/) the contents of the `build.spdx` Document created by `west spdx`; it will fail due to the missing checksum.

Note also the separate issue at #42070 which may cause other validation warnings, and which is being addressed as well.

**Impact**

The SPDX Document being created is not a valid SPDX Document, and might not be consumable by downstream tools that expect full conformance to the spec.

**Environment (please complete the following information):**

- Linux (Ubuntu 20.04.3 LTS)
- Zephyr SDK 0.13.2
- Zephyr main branch, commit d51a67b7c0
  - with one not-yet-committed change to zspdx/writer.py for #42070, which should have no relevance to this bug

**Additional context**

I'm not familiar enough with the full Zephyr build process to understand why the `libdrivers__interrupt_controller.a` file is no longer present once the build is complete. I don't know whether that reflects a separate bug.

However, regardless of why that's occurring, `west spdx` should be able to gracefully handle errors such as this, and should not generate invalid SPDX Documents even if files are unexpectedly missing.

**Comment by swinslow** [comment]
cc @kestewart for visibility

My current plan to address this is to modify the zspdx walker.py and/or scanner.py files so that a build file will be ignored if CMake reports it as being built, but it is not found on disk after the build is complete.

**Comment by mbolivar-nordic** [comment]
@swinslow any plans to address this in the near future?

**Comment by swinslow** [comment]
Hi @mbolivar-nordic, this should now be fixed with #43991. Non-existent files which were causing this bug will now be omitted from the SPDX document so that they will not cause it to be invalid due to no available checksum value.

@kestewart I added you as a second reviewer on that issue as well.
